openmc/mgxs/library.py

```diff
@@ -717,3 +721,267 @@ def load_from_file(filename='mgxs', directory='mgxs'):
+                    using_multiplicity = True
+                else:
+                    using_multiplicity = False
+
+                if using_multiplicity:
+                    nuscatt_mgxs = self.get_mgxs(domain,
+                                                 'nu-scatter matrix')
+                    xsdata.set_scatter_mgxs(nuscatt_mgxs, xs_type=xs_type,
+                                            nuclide=nuclide)
+                else:
+                    if 'nu-scatter matrix' in self.mgxs_types:
+                        nuscatt_mgxs = self.get_mgxs(domain,
+                                                     'nu-scatter matrix')
+                        xsdata.set_scatter_mgxs(nuscatt_mgxs, xs_type=xs_type,
+                                                nuclide=nuclide)
+
+                        # Since we are not using multiplicity, then
+                        # scattering multiplication (nu-scatter) must be
+                        # accounted for approximately by using an adjusted
+                        # absorption cross section.
+                        if 'total' in self.mgxs_types:
+                            xsdata._absorption = \
+                                np.subtract(xsdata.total,
+                                            np.sum(xsdata.scatter[0, :, :],
+                                                   axis=1))
+
+                xsdatas.append(xsdata)
+
+        # Add XSdatas to file
+        mgxs_file.add_xsdatas(xsdatas)
+
+        # Finally, write the file
+        mgxs_file.export_to_xml(full_filename)
+
+        if return_names:
+            return mat_names
+
+    def check_library_for_openmc_mgxs(self):
+        """This routine will check the MGXS Types within the provided
+        Library to ensure the data types provided can be used to create
+        a MGXS Library for OpenMC's Multi-Group mode via the
+        `Library.write_mg_library` method.
+        The rules to check include:
+        - Either total or transport should be present.
+          - Both can be available if one wants, but we should
+            use whatever corresponds to Library.correction (if P0: transport)
+        - Absorption and total (or transport) are required.
+        - A nu-fission cross section and chi values are not required as a
+          fixed source problem could be the target.
+        - Fission and kappa-fission are not required as they are only
+          needed to support tallies the user may wish to request.
+        - A nu-scatter matrix is required.
+          - Having both nu-scatter (of any order) and scatter
+            (at least isotropic) matrices is preferred
+          - If only nu-scatter, need total (not transport), to
+            be used in adjusting absorption
+            (i.e., reduced_abs = tot - nuscatt)
+
+        See also
+        --------
+        Library.write_mg_library(...)
+
+        """
+
+        error_flag = False
```

Comment:
Yeah, I do prefer the errors simply because I think there is no point in continuing if the tallies which were set up wont be sufficient.